### PR TITLE
[#1246] docs(lakehouse-iceberg): add Gravitino Iceberg rest catalog service blog

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
add Gravitino Iceberg rest catalog service blog

### Why are the changes needed?

Fix: #1246 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
just document

### CODE REVIEW

File: docs/iceberg-rest-service-blog.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+title: "Gravitino Iceberg REST catalog service"
+slug: /iceberg-rest-service-blog
+keywords:
+  - Iceberg REST catalog
+license: "Copyright 2023 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
+---
+
+## Background
+[Apache Iceberg](https://iceberg.apache.org/) is a popular high-performance format for huge analytic tables. Iceberg catalogs is responsible for creating, dropping, alerting Iceberg tables and namespaces. The catalog stores a reference to the current table metadata, and provides the grant of atomic update to the current table metadata. Iceberg catalog is completely pluggable and provides several implements:
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:11]: In Gravitino Iceberg catalog is responsible....
and alerting
"the grant of atomic update to the current table metadata." is not easily understood and needs to be rewritten
The Iceberg catalog...

File: docs/iceberg-rest-service-blog.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+1. Hive catalog, tracks namespaces and tables using a Hive metastore.
+2. JDBC catalog, tracks namespaces and tables in a simple JDBC database like MySQL or PostGreSQL.
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:13]: PostgreSQL

File: docs/iceberg-rest-service-blog.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+3. HADOOP catalog, tracks namespaces and tables in HDFS.
+ 
+All above catalogs implement all catalog logics in client side, challenges persist:
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:16]: Suggest:
All mentioned catalogs implement catalog logic on the client side, but some challenges exist:

File: docs/iceberg-rest-service-blog.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+1. **Client compatibility**: Client is responsible for compatibility.
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:17]: This is unclear what or who is the client here?

File: docs/iceberg-rest-service-blog.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+2. **Language and Ecosystem Compatibility**: The current Java implementation poses challenges for maintenance in other languages and native engines.
+3. **Consistency**: Consistency across implementations.
+4. **Performance Limitations**: Opportunities for performance enhancements remain underexplored.
+
+In response, the Iceberg 0.14.0 release ushered in the REST catalog, outlined in the [Iceberg REST API specification](https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml). This introduces a paradigm shift, placing the bulk of logic on the server side, making the client lightweight.
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:22]: ...the Iceberg 0.14.0 release supports a REST catalog, as outlined....
This places the bulk of the logic on the server side, making for lightweight clients.

File: docs/iceberg-rest-service-blog.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+[Gravitino](https://datastrato.ai/gravitino/) is opensource, high-performance, geo-distributed, and federated metadata lake, provides a unified interface to operate diverse catalogs like Hive, Iceberg, JDBC. Gravitino seamlessly integrates Iceberg REST interfaces defined in the Iceberg REST API specification, capable of serving as an Iceberg REST service.
+
+## Gravitino Iceberg REST catalog service
+
+Gravitino Iceberg REST service aspires to be a high-performance, product-quality Iceberg REST service, fully supporting the Iceberg REST interface.
+Architecture Overview:
+
+![Gravitino iceberg rest catalog Image](assets/gravitino-iceberg-rest-catalog-architecture.png)
+
+It's working in the proxy mode, proxy operations to the Iceberg backend catalog like `HiveCatalog` or `JdbcCatalog`. It covers all namespace and table operations, including creating tables, dropping tables, altering tables and renaming tables.
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:33]: altering tables, and

File: docs/iceberg-rest-service-blog.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+Beyond these fundamental operations, it provides server metrics and OAuth support crucial for production environments.
+
+## Roadmap
+
+1. **Security Enhancement**: Broadening support for various authentication and authorization systems.
+2. **Performance improvement**: Implementing cache methods to enhance operational efficiency.
+3. **Full interface support**: Store Iceberg metrics.
+
+If you are any ideas or requirements, feel free to open an issue in the [Gravitino GitHub repository](https://github.com/datastrato/gravitino).  Your contributions will propel the evolution of this dynamic Iceberg REST catalog service.
```
Comment (justinmclean) [inline docs/iceberg-rest-service-blog.md:42]: If you have any ideas...

Remove or reword "Your contribution will propel the evolution of this dynamic Iceberg REST catalog service."

**Review by justinmclean — COMMENTED** [review]
I am not sure what you are using to help generate the English, but it needs to be simpler and less flowery language-wise.

**Comment by FANNG1** [comment]
> I am not sure what you are using to help generate the English, but it needs to be simpler and less flowery language-wise.

yes, the doc is polished by chatGPT,  I modified the document in https://docs.google.com/document/d/18Ax9iswtCtx09shuPpW06O6aUaVH1K4w2g2Xsmpowm0/edit, please update the latest change in google docs and I'll close this pr.

**Comment by justinmclean** [comment]
The changes have not been made.

**Comment by FANNG1** [comment]
> The changes have not been made.

The changes are modified in https://docs.google.com/document/d/18Ax9iswtCtx09shuPpW06O6aUaVH1K4w2g2Xsmpowm0/edit,  Let's go to the Google doc?  It's easy to collaborate the work, every person can change to document.

**Comment by justinmclean** [comment]
I would suggest that the workflow be to have people review the Google doc first and have it approved before making a pull request?

**Comment by justinmclean** [comment]
Also, GitHub is public, but that Google link requires permission; the process needs to be modified to be more open.

**Comment by FANNG1** [comment]
> I would suggest that the workflow be to have people review the Google doc first and have it approved before making a pull request?

ok, let's goto google doc first.

**Comment by jerryshao** [comment]
I'm going to close this PR, as this PR should not exist in Gravitino repo.

**Pull request closed without merging**